### ISSUE TITLE
After update to 1.27.2,  cannot store metrics: errors occurred during rows addition

### ISSUE THREAD

**Issue by alexandrst88** [VictoriaMetrics/VictoriaMetrics#181]
**Describe the bug**
After, upgrade to victoria-metrics-20190905-092823-tags-v1.27.2-0-g88f8670e, we got a lot of these error messages.

> ep 10 14:21:43 ip-172-20-6-16 victoriametrics[8346]: 2019-09-10T14:21:43.437+0000        error        lib/httpserver/httpserver.go:425        error in "/api/v1/write": cannot store metrics: errors occurred during rows addition: cannot obtain TSID for MetricName "traefik_backend_request_duration_seconds_bucket\x01namespace\x01kube-system\x01service\x01traefik-ingress-external-addon\x01job\x01traefik-ingress-external-addon\x01instance\x01100.96.20.67:8080\x01pod\x01traefik-ingress-external-addon-67ccc9b477-l46qt\x01backend\x011-example.com/\x01code\x01403\x01endpoint\x018080\x01le\x010.1\x01method\x01GET\x01prometheus\x01monitoring/prometheus-operator-prometheus\x01prometheus_replica\x01prometheus-prometheus-operator-prometheus-0\x01protocol\x01http\x01": cannot search TSID by MetricName "traefik_backend_request_duration_seconds_bucket\x01namespace\x01kube-system\x01service\x01traefik-ingress-external-addon\x01job\x01traefik-ingress-external-addon\x01instance\x01100.96.20.67:8080\x01pod\x01traefik-ingress-external-addon-67ccc9b477-l46qt\x01backend\x011-example.com/\x01code\x01403\x01endpoint\x018080\x01le\x010.1\x01method\x01GET\x01prometheus\x01monitoring/prometheus-operator-prometheus\x01prometheus_replica\x01prometheus-prometheus-operator-prometheus-0\x01protocol\x01http\x01": error when searching TSID by metricName; searchPrefix "\x00traefik_backend_request_duration_seconds_bucket\x01namespace\x01kube-system\x01service\x01traefik-ingress-external-addon\x01job\x01traefik-ingress-external-addon\x01instance\x01100.96.20.67:8080\x01pod\x01traefik-ingress-external-addon-67ccc9b477-l46qt\x01backend\x011-example.com/\x01code\x01403\x01endpoint\x018080\x01le\x010.1\x01method\x01GET\x01prometheus\x01monitoring/prometheus-operator-prometheus\x01prometheus_replica\x01prometheus-prometheus-operator-prometheus-0\x01protocol\x01http\x01\x02": cannot seek "\x00rabbitmq_queue_auto_delete\x01namespace\x01prod\x01job\x01rmq-cluster\x01type\x01classic\x01instance\x01rmq.prod.example.internal:15672\x01cluster_name\x01rmq\x01policy\x01ha-all\x01prometheus\x01monitoring/prometheus-operator-prometheus

**To Reproduce**
Update from 1.27.0 to 1.27.2

**Expected behavior**
No error messages in logs :)

**Version**
The line returned when passing `--version` command line flag to binary. For example:
```
$ ./victoria-metrics-prod --version
victoria-metrics-20190905-092823-tags-v1.27.2-0-g88f8670e
```

**Additional context**
N/A.

**Comment by alexandrst88** [comment]
```2019-09-10T14:48:58.618+0000	error	lib/httpserver/httpserver.go:425	error in "/api/v1/write": cannot store metrics: errors occurred during rows addition: cannot obtain TSID for MetricName "rabbitmq_queue_consumer_utilisation\x01namespace\x01bar\x01job\x01rmq-cluster\x01type\x01classic\x01instance\x01rmq.bar.example.internal:15672\x01cluster_name\x01rmq\x01policy\x01ha-all\x01prometheus\x01monitoring/prometheus-operator-prometheus\x01prometheus_replica\x01prometheus-prometheus-operator-prometheus-0\x01queue\x01foo.journey-service.applications\x01vhost\x01/\x01": cannot search TSID by MetricName "rabbitmq_queue_consumer_utilisation\x01namespace\x01bar\x01job\x01rmq-cluster\x01type\x01classic\x01instance\x01rmq.bar.example.internal:15672\x01cluster_name\x01rmq\x01policy\x01ha-all\x01prometheus\x01monitoring/prometheus-operator-prometheus\x01prometheus_replica\x01prometheus-prometheus-operator-prometheus-0\x01queue\x01foo.journey-service.applications\x01vhost\x01/\x01": error when searching TSID by metricName; searchPrefix "\x00rabbitmq_queue_consumer_utilisation\x01namespace\x01bar\x01job\x01rmq-cluster\x01type\x01classic\x01instance\x01rmq.bar.example.internal:15672\x01cluster_name\x01rmq\x01policy\x01ha-all\x01prometheus\x01monitoring/prometheus-operator-prometheus\x01prometheus_replica\x01prometheus-prometheus-operator-prometheus-0\x01queue\x01foo.journey-service.applications\x01vhost\x01/\x01\x02": cannot seek "\x00rabbitmq_queue_consumer_utilisation\x01namespace\x01bar\x01job\x01rmq-cluster\x01type\x01classic\x01instance\x01rmq.bar.example.internal:15672\x01cluster_name\x01rmq\x01policy\x01ha-all\x01prometheus\x01monitoring/prometheus-operator-prometheus\x01prometheus_replica\x01prometheus-prometheus-operator-prometheus-0\x01queue\x01foo.journey-service.applications\x01vhost\x01/\x01\x02": cannot get index block: cannot unmarshal block headers from index block (offset=27880151, size=3841): block headers must be sorted by firstItem; unmarshaled unsorted block headers: []mergeset.blockHeader{mergeset.blockHeader{commonPrefix:[]uint8{0x0, 0x72, 0x61, 0x62, 0x62, 0x69, 0x74, 0x6d, 0x71, 0x5f, 0x71, 0x75, 0x65, 0x75, 0x65, 0x5f, 0x61, 0x75, 0x74, 0x6f, 0x5f, 0x64, 0x65, 0x6c, 0x65, 0x74, 0x65, 0x1, 0x6e, 0x61, 0x6d, 0x65, 0x73, 0x70, 0x61, 0x63, 0x65, 0x1, 0x70, 0x72, 0x6f, 0x64, 0x1, 0x6a, 0x6f, 0x62, 0x1, 0x72, 0x6d, 0x71, 0x2d, 0x63, 0x6c, 0x75, 0x73, 0x74, 0x65, 0x72, 0x1, 0x74, 0x79, 0x70, 0x65, 0x1, 0x63, 0x6c, 0x61, 0x73, 0x73, 0x69, 0x63, 0x1, 0x69, 0x6e, 0x73, 0x74, 0x61, 0x6e, 0x63, 0x65, 0x1, 0x72, 0x6d, 0x71, 0x2e, 0x70, 0x72, 0x6f, 0x64, 0x2e, 0x64, 0x72, 0x65, 0x61, 0x6d, 0x74, 0x65, 0x61, 0x6d, 0x2e, 0x69, 0x6e, 0x74, 0x65, 0x72, 0x6e, 0x61, 0x6c, 0x3a, 0x31, 0x35, 0x36, 0x37, 0x32, 0x1, 0x63, 0x6c, 0x75, 0x73, 0x74, 0x65, 0x72, 0x5f, 0x6e, 0x61, 0x6d, 0x65, 0x1, 0x72, 0x6d, 0x71, 0x1, 0x70, 0x6f, 0x6c, 0x69, 0x63, 0x79, 0x1, 0x68, 0x61, 0x2d, 0x61, 0x6c, 0x6c, 0x1, 0x70, 0x72, 0x6f, 0x6d, 0x65, 0x74, 0x68, 0x65, 0x75, 0x73, 0x1, 0x6d, 0x6f, 0x6e, 0x69, 0x74, 0x6f, 0x72, 0x69, 0x6e, 0x67, 0x2f, 0x70, 0x72, 0x6f, 0x6d, 0x65, 0x74, 0x68, 0x65, 0x75, 0x73, 0x2d, 0x6f, 0x70, 0x65, 0x72, 0x61, 0x74, 0x6f, 0x72, 0x2d, 0x70, 0x72, 0x6f, 0x6d, 0x65, 0x74, 0x68, 0x65, 0x75, 0x73, 0x1, 0x70, 0x72, 0x6f, 0x6d, 0x65, 0x74, 0x68, 0x65, 0x75, 0x73, 0x5f, 0x72, 0x65, 0x70, 0x6c, 0x69, 0x63, 0x61, 0x1, 0x70, 0x72, 0x6f, 0x6d, 0x65, 0x74, 0x68, 0x65, 0x75, 0x73, 0x2d, 0x70, 0x72, 0x6f, 0x6d, 0x65, 0x74, 0x68, 0x65, 0x75, 0x73, 0x2d, 0x6f, 0x70, 0x65, 0x72, 0x61, 0x74, 0x6f, 0x72, 0x2d, 0x70, 0x72, 0x6f, 0x6d, 0x65, 0x74, 0x68, 0x65, 0x75, 0x73, 0x2d, 0x30, 0x1, 0x71, 0x75, 0x65, 0x75, 0x65, 0x1, 0x67, 0x61, 0x6d, 0x65, 0x73, 0x2e, 0x63}, firstItem:[]uint8{0x0, 0x72, 0x61, 0x62, 0x62, 0x69, 0x74, 0x6d, 0x71, 0x5f, 0x71, 0x75, 0x65, 0x75, 0x65, 0x5f, 0x61, 0x75, 0x74, 0x6f, 0x5f, 0x64, 0x65, 0x6c, 0x65, 0x74, 0x65, 0x1, 0x6e, 0x61, 0x6d, 0x65, 0x73, 0x70, 0x61, 0x63, 0x65, 0x1, 0x70, 0x72, 0x6f, 0x64, 0x1, 0x6a, 0x6f, 0x62, 0x1, 0x72, 0x6d, 0x71, 0x2d, 0x63, 0x6c, 0x75, 0x73, 0x74, 0x65, 0x72, 0x1, 0x74, 0x79, 0x70, 0x65, 0x1, 0x63, 0x6c, 0x61, 0x73, 0x73, 0x69, 0x63, 0x1, 0x69, 0x6e, 0x73, 0x74, 0x61, 0x6e, 0x63, 0x65, 0x1, 0x72, 0x6d, 0x71, 0x2e, 0x70, 0x72, 0x6f, 0x64, 0x2e, 0x64, 0x72, 0x65, 0x61, 0x6d, 0x74, 0x65, 0x61, 0x6d, 0x2e, 0x69, 0x6e, 0x74, 0x65, 0x72, 0x6e, 0x61, 0x6c, 0x3a, 0x31, 0x35, 0x36, 0x37, 0x32, 0x1, 0x63, 0x6c, 0x75, 0x73, 0x74, 0x65, 0x72, 0x5f, 0x6e, 0x61, 0x6d, 0x65, 0x1, 0x72, 0x6d, 0x71, 0x1, 0x70, 0x6f, 0x6c, 0x69, 0x63, 0x79, 0x1, 0x68, 0x61, 0x2d, 0x61, 0x6c, 0x6c, 0x1, 0x70, 0x72, 0x6f, 0x6d, 0x65, 0x74, 0x68, 0x65, 0x75, 0x73, 0x1, 0x6d, 0x6f, 0x6e, 0x69, 0x74, 0x6f, 0x72, 0x69, 0x6e, 0x67, 0x2f, 0x70, 0x72, 0x6f, 0x6d, 0x65, 0x74, 0x68, 0x65, 0x75, 0x73, 0x2d, 0x6f, 0x70, 0x65, 0x72, 0x61, 0x74, 0x6f, 0x72, 0x2d, 0x70, 0x72, 0x6f, 0x6d, 0x65, 0x74, 0x68, 0x65, 0x75, 0x73, 0x1, 0x70, 0x72, 0x6f, 0x6d, 0x65, 0x74, 0x68, 0x65, 0x75, 0x73, 0x5f, 0x72, 0x65, 0x70, 0x6c, 0x69, 0x63, 0x61, 0x1, 0x70, 0x72, 0x6f, 0x6d, 0x65, 0x74, 0x68, 0x65, 0x75, 0x73, 0x2d, 0x70, 0x72, 0x6f, 0x6d, 0x65, 0x74, 0x68, 0x65, 0x75, 0x73, 0x2d, 0x6f, 0x70, 0x65, 0x72, 0x61, 0x74, 0x6f, 0x72, 0x2d, 0x70, 0x72, 0x6f, 0x6d, 0x65, 0x74, 0x68, 0x65, 0x75, 0x73, 0x2d, 0x30, 0x1, 0x71, 0x75, 0x65, 0x75, 0x65, 0x1, 0x67, 0x61, 0x6d, 0x65, 0x73, 0x2e, 0x63, 0x68, 0x61, 0x6c, 0x6c, 0x65, 0x6e, 0x67, 0x65, 0x2d, 0x73, 0x65, 0x72, 0x76, 0x69, 0x63, 0x65, 0x2e, 0x63, 0x68, 0x61, 0x6c, 0x6c, 0x65, 0x6e, 0x67, 0x65, 0x5f, 0x66, 0x69, 0x6e, 0x61, 0x6c, 0x69, 0x7a, 0x61, 0x74, 0x69, 0x6f, 0x6e, 0x5f, 0x73, 0x63, 0x68, 0x65, 0x64, 0x75, 0x6c, 0x65, 0x64, 0x2e, 0x64, 0x65, 0x6c, 0x61, 0x79, 0x65, 0x64, 0x2e, 0x33, 0x36, 0x30, 0x30, 0x2e, 0x73, 0x65, 0x63, 0x1, 0x76, 0x68, 0x6f, 0x73, 0x74, 0x1, 0x2f, 0x1, 0x2, 0x12, 0xe0, 0xfb, 0xd0, 0xe9, 0x28, 0xc9, 0x1f, 0xc9, 0xbb, 0x36, 0x28, 0x6a, 0x5a, 0x8e, 0xf3, 0x15, 0xbd, 0x44, 0xf3, 0xd6, 0xdf, 0xa0, 0xf8}, marshalType:0x1, itemsCount:0xb5, itemsBlockOffset:0x3fba94fb, lensBlockOffset:0x20d6fe6, itemsBlockSize:0x246, lensBlockSize:0x78}, mergeset.blockHeader{commonPrefix:[]uint8{0x0, 0x72, 0x61, 0x62, 0x62, 0x69, 0x74, 0x6d, 0x71, 0x5f, 0x71, 0x75, 0x65, 0x75, 0x65, 0x5f, 0x61, 0x75, 0x74, 0x6f, 0x5f, 0x64, 0x65, 0x6c, 0x65, 0x74, 0x65, 0x1, 0x6e, 0x61, 0x6d, 0x65, 0x73, 0x70, 0x61, 0x63, 0x65, 0x1, 0x70, 0x72, 0x6f, 0x64, 0x1, 0x6a, 0x6f, 0x62, 0x1, 0x72, 0x6d, 0x71, 0x2d, 0x63, 0x6c, 0x75, 0x73, 0x74, 0x65, 0x72, 0x1, 0x74, 0x79, 0x70, 0x65, 0x1, 0x63, 0x6c, 0x61, 0x73, 0x73, 0x69, 0x63, 0x1, 0x69, 0x6e, 0x73, 0x74, 0x61, 0x6e, 0x63, 0x65, 0x1, 0x72, 0x6d, 0x71, 0x2e, 0x70, 0x72, 0x6f, 0x64, 0x2e, 0x64, 0x72, 0x65, 0x61, 0x6d, 0x74, 0x65, 0x61, 0x6d, 0x2e, 0x69, 0x6e, 0x74, 0x65, 0x72, 0x6e, 0x61, 0x6c, 0x3a, 0x31, 0x35, 0x36, 0x37, 0x32, 0x1, 0x63, 0x6c, 0x75, 0x73, 0x74, 0x65, 0x72, 0x5f, 0x6e, 0x61, 0x6d, 0x65, 0x1, 0x72, 0x6d, 0x71, 0x1, 0x70, 0x6f, 0x6c, 0x69, 0x63, 0x79, 0x1, 0x68, 0x61, 0x2d, 0x61, 0x6c, 0x6c, 0x1, 0x70, 0x72, 0x6f, 0x6d, 0x65, 0x74, 0x68, 0x65, 0x75, 0x73, 0x1, 0x6d, 0x6f, 0x6e, 0x69, 0x74, 0x6f, 0x72, 0x69, 0x6e, 0x67, 0x2f, 0x70, 0x72, 0x6f, 0x6d, 0x65, 0x74, 0x68, 0x65, 0x75, 0x73, 0x2d, 0x6f, 0x70, 0x65, 0x72, 0x61, 0x74, 0x6f, 0x72, 0x2d, 0x70, 0x72, 0x6f, 0x6d, 0x65, 0x74, 0x68, 0x65, 0x75, 0x73, 0x1, 0x70, 0x72, 0x6f, 0x6d, 0x65,```

**Comment by hatemosphere** [comment]
we just checked and it reproduces on 1.27.1 1.27.2, 1.27.0 doesn't have this issue

**Comment by valyala** [comment]
@alexandrst88 , thanks for the bug report. The issue should be fixed in the commit 0c8057924f34de84bc348a86d433ce86eeeec7ab . Try building from this commit and verifying whether the issue is fixed. See [build instructions](https://github.com/VictoriaMetrics/VictoriaMetrics/blob/master/README.md#how-to-build-from-sources).

**Comment by valyala** [comment]
The issue must be fixed in [v1.27.3](https://github.com/VictoriaMetrics/VictoriaMetrics/releases). @alexandrst88 , could you confirm this?

**Comment by rzhylak** [comment]
Hey. I had the same problem and I was able to reproduce it on version 1.27.0, and with version 1.26.0 I constantly saw VM container restarts during startup. As a workaround, I had to create a new storage directory and lost my historical time series. Just upgrade all my victoria-metrics instances to version 1.27.3 and the problem goes away. Logs are now pretty clear, and historical data also works well, thanks for quickly fixing this bug.
